### PR TITLE
ci: only run dawn slices views on main

### PR DESCRIPTION
**Pull request by wdconinc**
### Briefly, what does this PR introduce?
We are not able to keep up with workflows at times. While it is useful to make sure all jobs run, even on pull requests, it is more useful to make sure the visualization jobs run on main. This restricts prim file generation and dawn views to main only.

### What kind of change does this PR introduce?
- [ ] Bug fix (issue #__)
- [ ] New feature (issue #__)
- [ ] Documentation update
- [x] Other: __

### Please check if this PR fulfills the following:
- [ ] Tests for the changes have been added
- [ ] Documentation has been added / updated
- [ ] Changes have been communicated to collaborators

### Does this PR introduce breaking changes? What changes might users need to make to their code?
No.

### Does this PR change default behavior?
No.

### CODE REVIEW

**Review by veprbl — APPROVED** [review]
I agree

**Comment by c-dilks** [comment]
I like the speedup, but having `view1_top` is very useful when comparing a geometry-changing dev branch to `main` to see how things move, especially when `definitions.xml` is modified.

**Comment by c-dilks** [comment]
Can we keep `view1` for PRs? I also don't mind commenting out these `if: github.ref` lines temporarily the next time someone asks me to touch `definitions.xml`.

**Comment by wdconinc** [comment]
I don't find a reference for jobs.<job>.strategy.parallel.if support. We can use a quick job to create a list of views and tweak that based on PR vs main.

File: .github/workflows/linux-eic-shell.yml [hunk 1]
```diff
@@ -508,6 +508,7 @@ jobs:
   generate-prim-file:
     runs-on: ubuntu-latest
     needs: build
+    if: github.ref == 'refs/heads/main'
```
Comment (wdconinc) [inline .github/workflows/linux-eic-shell.yml:511]:
```suggestion
```

File: .github/workflows/linux-eic-shell.yml [hunk 2]
```diff
@@ -536,6 +537,7 @@ jobs:
   dawn-view:
     runs-on: ubuntu-latest
     needs: generate-prim-file
+    if: github.ref == 'refs/heads/main'
```
Comment (wdconinc) [inline .github/workflows/linux-eic-shell.yml:540]:
```suggestion
```

**Review by wdconinc — COMMENTED** [review]
How about we just skip the dawn slices jobs? Those are largely responsible too. It would be a start and we can continue to evaluate.

**Review by c-dilks — APPROVED** [review]

**Pull request merged**